docs/docs/backup-restore.md

```diff
@@ -0,0 +1,118 @@
+This method is marked experimental - the API endpoint may change in the future.
+
+- Executable packaged with Marathon
+The marathon native package ships with a start-script that can be used to create a backup.
+Please note: this command will connect directly to zk and backup the state from the defined location. 
+Use this method only, if there is no Marathon instance running!
+
+
+```
+$> marathon -main mesosphere.marathon.core.storage.backup.Backup --help
+$> marathon -main mesosphere.marathon.core.storage.backup.Backup --backup_location {backup_uri} --zk {zk_uri}
+```
+
+
+## Restore
+
+A backup that has been created, can be restored. Every restore operation will perform the following steps:
+- the current state in zk under the defined path will be removed
+- the state from the backup will be restored to the defined zk path 
+- the next Marathon leader will migrate this state to the current version
+
+### Options to restore from a backup 
+
+There are multiple ways to restore a backup:
+
+- API endpoint `DELETE /v2/leader?restore={URI}` (experimental)
+It is possible to instruct a running marathon instance to restore its state from a backup.
+Using this endpoint will force the current running leader to abdicate.
+The newly elected leader will perform a clean and restore operation, do a migration (if needed) and start up.
+This method is marked experimental - the API endpoint may change in the future.
+
+- Executable packaged with Marathon
+The marathon native package ships with a start-script that can be used to restore a backup.
+Please note: this command will connect directly to zk and restore the state from the defined location. 
+Use this method only, if there is no Marathon instance running!
+ 
+ 
+```
+$> marathon -main mesosphere.marathon.core.storage.backup.Restore --help
+$> marathon -main mesosphere.marathon.core.storage.backup.Restore --backup_location {backup_uri} --zk {zk_uri}
+```
+
+
+## Available Backup Storage Provider
+
+The backup & restore functionality allows different storage providers to save and load a backup.
+Currently, `file` and `s3` providers are implemented. 
+If you need additional providers, please create a feature request in our [issue tracker](https://jira.mesosphere.com/secure/CreateIssue!default.jspa?pid=10401)
+  
+### File storage
+This provider uses the file system to store and load a backup.
+It uses the default file URI syntax.
+
+Example: `file:///path/to/file`
+
+Attention: do not use the local file system if you are running a multi-master setup or the leading master can run on different machines.
+The backup and restore operation could be performed on different machines and the backup file might not be accessible.
+Use a shared file system to use the file storage provider in such a setup.
+
+### S3 storage
+This provider will use the Amazon S3 storage to store and load a backup.
+An S3 location is defined with this URI syntax (Also see [s3tools](http://s3tools.org/s3_about))
```

Comment:
"Define an S3 location with the following URI syntax. See [S3 tools](http://s3tools.org/s3_about) for more information.